### PR TITLE
initial version of code for population simulation

### PR DESCRIPTION
**Pull request by KatrinCoboeken**

### CODE REVIEW

File: Code/engine_library/addDosetablePerWeight.m [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+%           - BWmax	 (double vector) [kg] vector which defines upper limit of one body weight bin
+%           - targetParameter (double vector) values, which the target parameter will be in the corresponding body weight bin
+%
+% Outputs:
+%   - parPathes (cellarray):  pathnames of the population parameters,
+%                   changed acoording to study desing
+%   - parValues (double matrix):  values of the population parameters and
+%                   individuals, changed acoording to study desing
+
+% Open Systems Pharmacology Suite;  support@systems-biology.com
```
Comment (msevestre) [inline Code/engine_library/addDosetablePerWeight.m:22]: Should not we remove email `support@systems-biology.com`  everywhere?

File: Code/engine_library/addDosetablePerWeight.m [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function addDosetablePerWeight(Settings,targetParameterList,parPathes,parValues,SplitCondition)
+% add new lines to the population, which sets
+%
+% Inputs:
+%   - Settings       (structure) containing global settings see GETDEFAULTWORKFLOWSETTINGS
+%   - targetParameterList (cellarray of string)   List of
+%                       applicationparameter, which should be added whit adjusted dose to the
+%                       population values
+%   - parPathes (cellarray):  pathnames of the population parameters
```
Comment (msevestre) [inline Code/engine_library/addDosetablePerWeight.m:9]: `Paths` not `Pathes`

File: Code/engine_library/addDosetablePerWeight.m [hunk 3]
```diff
@@ -0,0 +1,30 @@
+function addDosetablePerWeight(Settings,targetParameterList,parPathes,parValues,SplitCondition)
+% add new lines to the population, which sets
+%
+% Inputs:
+%   - Settings       (structure) containing global settings see GETDEFAULTWORKFLOWSETTINGS
+%   - targetParameterList (cellarray of string)   List of
+%                       applicationparameter, which should be added whit adjusted dose to the
```
Comment (msevestre) [inline Code/engine_library/addDosetablePerWeight.m:7]: with

File: Code/engine_library/Contents.m [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+%
+%  functions, which reads input files
+%   readPopulationCSV                           - reads a poulation csv file 
+%   readPopulationResultfile                    - reads a simulation result file exported by PK-SIM
+%   readOutputSelection                         - reads list of selected outputs from csv file
+%
+%  internal functions
+%   initializeWorkflow                          - initialise the workflow, sets up the logfile, and sets global settings
+%   getApplicationProtocollFromXML              - get properties of the applicationsprotocol
+%   calculatePKParameter_forApplicationProtocol - calculates PK Parameter, same List as in PK-Sim, for multi or single application
```
Comment (msevestre) [inline Code/engine_library/Contents.m:23]: _ in name is unusual no? Why not calculatePKParameterForApplicationProtocol?

File: Code/engine_library/getApplicationProtocollFromXML.m [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    end
+    if existsParameter(sprintf('*Application_%d|ProtocolSchemaItem|Dose',iApp),simulationIndex,'parametertype','readonly');
+        ApplicationProtocol(iApp).dose = getParameter(sprintf('*Application_%d|ProtocolSchemaItem|Dose',iApp),simulationIndex,'parametertype','readonly');
+        ApplicationProtocol(iApp).dose =  ApplicationProtocol(iApp).dose; 
+        ApplicationProtocol(iApp).isDosePerBodyweight = getParameter(sprintf('*Application_%d|ProtocolSchemaItem|Dose',iApp),simulationIndex,'parametertype','readonly','property','isFormula');
+    end
+    if existsParameter(sprintf('*Application_%d|ProtocolSchemaItem|DrugMass',iApp),simulationIndex,'parametertype','readonly');
+        ApplicationProtocol(iApp).drugMass = getParameter(sprintf('*Application_%d|ProtocolSchemaItem|DrugMass',iApp),simulationIndex,'parametertype','readonly');
+    end
+    if existsParameter(sprintf('*Application_%d|ProtocolSchemaItem|DosePerBodyWeight',iApp),simulationIndex,'parametertype','readonly');
```
Comment (msevestre) [inline Code/engine_library/getApplicationProtocollFromXML.m:56]: We have just added the BSA dosing. This probably needs adjustment here

File: Code/engine_library/addDosetablePerWeight.m (hunk 1 above)
Comment (Yuri05) [reply Code/engine_library/addDosetablePerWeight.m:22]: Yep, replace with http://forum.open-systems-pharmacology.org

File: Code/engine_library/Contents.m (hunk 4 above)
Comment (KatrinCoboeken) [reply Code/engine_library/Contents.m:23]: You are right. I will change it.

**Comment by msevestre** [comment]
A quick question to everyone here so that I understand the thought process:

Why is the reporting component aware of **how** to calculate PK-Parameters or simulation population  etc.. ? In my opinion, we are duplicating the knowledge once again and this will lead to serious headache down the line. 

I would strongly advise to have the component use the existing interfaces (which may need to be enhanced?) to run the simulations and calculate PK-Parameters instead of reinventing the wheel. Implementation will run out of sync otherwise (see my comment on BSA: code is brand new and already outdated)

**Comment by msevestre** [comment]
Closing because of change in code folder structure. Juri will explain :)

**Pull request closed without merging**